src/Core/src/Platform/Android/EditTextExtensions.cs

```diff
@@ -320,11 +320,18 @@ internal static void SetInputType(this EditText editText, ITextInput textInput)
 			editText.SetSelection(previousCursorPosition);
 		}
 
-		internal static bool IsCompletedAction(this EditorActionEventArgs e, ImeAction? currentInputImeFlag)
+		internal static bool IsCompletedAction(this EditorActionEventArgs e, ImeAction currentInputImeFlag)
 		{
 			var actionId = e.ActionId;
 			var evt = e.Event;
 
+			// On API 34 it looks like they fixed the issue where the actionId is ImeAction.ImeNull when using a keyboard
+			// so I'm just setting the actionId here to whatever the user has 
+			if (actionId == ImeAction.ImeNull && evt?.KeyCode == Keycode.Enter)
+			{
+				actionId = currentInputImeFlag;
+			}
+
```

Comment:
So for pre api34 now - if we check the actionId on the entrys, after the action is completed, it should no longer say the actionId is null and instead show the expected one?

Reply:
still just a bit confused because that's my interpretatin of your comment and also because I'm seeing this code both here, and in EntryHandler.Android.cs